rslib/src/import_export/package/apkg/import/media.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright: Ankitects Pty Ltd and contributors
+// License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+
+use std::{collections::HashMap, fs::File, mem};
+
+use zip::ZipArchive;
+
+use super::Context;
+use crate::{
+    import_export::{
+        package::{
+            media::{extract_media_entries, SafeMediaEntry},
+            Meta,
+        },
+        ImportProgress,
+    },
+    media::{
+        files::{add_hash_suffix_to_file_stem, sha1_of_reader},
+        MediaManager,
+    },
+    prelude::*,
+};
+
+/// Map of source media files, that do not already exist in the target.
+#[derive(Default)]
+pub(super) struct MediaUseMap {
+    /// original, normalized filename → (refererenced on import material,
+    /// entry with possibly remapped filename)
+    checked: HashMap<String, (bool, SafeMediaEntry)>,
+    /// Static files (latex, underscored). Usage is not tracked, and if the name
+    /// already exists in the target, it is skipped regardless of content equality.
+    unchecked: Vec<SafeMediaEntry>,
+}
+
+struct ProgressHandler<F: FnMut(usize) -> Result<()>> {
+    progress_fn: F,
+    counter: usize,
+}
+
+impl<F: FnMut(usize) -> Result<()>> ProgressHandler<F> {
+    fn new(progress_fn: F) -> Self {
+        Self {
+            progress_fn,
+            counter: 0,
+        }
+    }
+
+    fn increment(&mut self) -> Result<()> {
+        self.counter += 1;
+        if self.counter % 17 == 0 {
```

Comment:
I've copied this number in https://github.com/ankitects/anki/pull/1743/commits/23eda19677eb880f6468370e11c492f425c9d144 - I presume you did this so the rightmost digit is not always zero?

Reply:
Yes, looks more dynamic to me. Also, the number might be a bit low. I'll probably genralise `ProgressHandler`, so it can be used at some other places as well, and make the frequence customisable.